server/src/main/kotlin/suwayomi/tachidesk/manga/impl/update/Updater.kt

```diff
@@ -0,0 +1,72 @@
+package suwayomi.tachidesk.manga.impl.update
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import mu.KotlinLogging
+import suwayomi.tachidesk.manga.impl.Chapter
+import suwayomi.tachidesk.manga.model.dataclass.MangaDataClass
+
+class Updater : IUpdater {
+    private val logger = KotlinLogging.logger {}
+    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
+
+    private var isRunning: Boolean = false
+
+    private var tracker = HashMap<String, UpdateJob>()
+    private var updateChannel = Channel<UpdateJob>()
+    private val statusChannel = MutableStateFlow(UpdateStatus())
+    private var updateJob: Job? = null
+
+    init {
+        updateJob = createUpdateJob()
+    }
+
+    private fun createUpdateJob(): Job {
+        return scope.launch {
+            while (true) {
+                val job = updateChannel.receive()
+                process(job)
+                isRunning = !updateChannel.isEmpty
+                val value = UpdateStatus(tracker, isRunning)
+                statusChannel.emit(value)
```

Comment:
```suggestion
                statusChannel.value = value
```
Assigning the stateflows state, which will automatically emit if the old value and the new value is not equal